src/Avalonia.Input/InputElement.cs

```diff
@@ -516,9 +516,12 @@ protected override void OnAttachedToVisualTreeCore(VisualTreeAttachmentEventArgs
         /// <param name="e">The event args.</param>
         protected virtual void OnGotFocus(GotFocusEventArgs e)
         {
-            var isFocused = e.Source == this;
-            _isFocusVisible = isFocused && (e.NavigationMethod == NavigationMethod.Directional || e.NavigationMethod == NavigationMethod.Tab);
-            IsFocused = isFocused;
+            if (IsEnabled & IsEffectivelyEnabled)
+            {
+                var isFocused = e.Source == this;
+                _isFocusVisible = isFocused && (e.NavigationMethod == NavigationMethod.Directional || e.NavigationMethod == NavigationMethod.Tab);
+                IsFocused = isFocused;
+            }
```

Comment:
Don't think it's correct change.
As I understand disabled elements still get focused (I haven't tested it to confirm), which means with this change old focused element will be unfocused, new element will be focused internally, but only IsFocused property won't be updated.
Probably you need to take a look at FocusManager class, there should be logic to decide if element should be focused.

Reply:
Or actually, it shouldn't get focus when it's disabled. #6957 problem is about disabling element **while** it is focused. So, this changes block is not needed.

We might want to unfocused element when it gets disabled, but for this PR I think Button.cs changes are enough. Also it gets complicated like what exactly should be focused instead, should be compared with WPF/UWP.